cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/InMemoryConfigurator.java

```diff
@@ -79,17 +80,19 @@
   private String appClassName;
   private Id.Artifact artifactId;
 
-  public InMemoryConfigurator(CConfiguration cConf, Id.Artifact artifactId, String appClassName,
+  public InMemoryConfigurator(CConfiguration cConf, Id.Namespace namespace, Id.Artifact artifactId, String appClassName,
```

Comment:
Hmm, wondering the field names can be better. For example, what is the difference between ``namespace`` and ``artifactId.getNamespaceId()`` ? Without context, ``namespace`` variable seems redundant.